table/table.go

```diff
@@ -154,6 +160,8 @@ type Table interface {
 
 	// Type returns the type of table
 	Type() Type
+
+	LocationPartition(ctx sessionctx.Context, r []types.Datum) (int, error)
```

Comment:
`LocationPartition` should not be put in the `Table` interface.
There is a `PartitionedTable` for it.